R/seasonal_onset.R

```diff
@@ -5,34 +5,25 @@
 #'
-#' @param tsd An object containing time series data with 'time' and 'observation.'
-#' @param k An integer specifying the window size for modeling growth rates.
-#' @param level The confidence level for parameter estimates, a numeric value between 0 and 1.
-#' @param disease_threshold An integer specifying the threshold for considering a disease outbreak. It defines
-#' the per time-step disease threshold that has to be surpassed to possibly trigger a seasonal onset alarm.
-#' If the total number of cases in a window of size k exceeds `disease_threshold * k`, a seasonal onset alarm
-#' can be triggered.
-#' @param family A character string specifying the family for modeling. Choose between "poisson," or "quasipoisson".
+#' @param tsd `r rd_tsd()`
+#' @param k An integer specifying the window size for modeling growth rates for the onset.
+#' @param level The confidence level for onset parameter estimates, a numeric value between 0 and 1.
+#' @param disease_threshold `r rd_disease_threshold(usage = "onset")`
+#' @param family `r rd_family()`
 #' @param na_fraction_allowed Numeric value between 0 and 1 specifying the fraction of observables in the window
-#' of size k that are allowed to be NA.
-#' @param season_weeks A numeric vector of length 2, `c(start,end)`, with the start and end weeks of the seasons to
-#' stratify the observations by. Must span the new year; ex: `season_weeks = c(21, 20)`. Default, `NULL`, is no
-#' stratification by season.
+#' of size k that are allowed to be NA in onset calculations.
+#' @param season_weeks `r rd_season_weeks(usage = "onset")`
 #'
 #' @return A `seasonal_onset` object containing:
 #'   - 'reference_time': The time point for which the growth rate is estimated.
 #'   - 'observation': The observation in the reference time point.
 #'   - 'season': The stratification of observables in corresponding seasons.
 #'   - 'growth_rate': The estimated growth rate.
-#'   - 'lower_growth_rate': The lower bound of the growth rate's confidence
-#'   interval.
-#'   - 'upper_growth_rate': The upper bound of the growth rate's confidence
-#'   interval.
-#'   - 'growth_warning': Logical. Is the growth rate significantly higher than
-#'   zero?
+#'   - 'lower_growth_rate': The lower bound of the growth rate's confidence interval.
+#'   - 'upper_growth_rate': The upper bound of the growth rate's confidence interval.
+#'   - 'growth_warning': Logical. Is the growth rate significantly higher than zero?
 #'   - 'sum_of_cases': The sum of cases within the time window.
-#'   - 'sum_of_cases_warning': Logical. Does the Sum of Cases exceed the
-#'   disease threshold?
+#'   - 'sum_of_cases_warning': Logical. Does the Sum of Cases exceed the disease threshold?
 #'   - 'seasonal_onset_alarm': Logical. Is there a seasonal onset alarm?
 #'   - 'skipped_window': Logical. Was the window skipped due to missing?
 #'   - 'converged': Logical. Was the IWLS judged to have converged?
```

Comment:
This could also be made a template that you could reuse in your combined_seasonal_output function.

Template:
```
rd_season_onset_return <- paste(
  "A `seasonal_onset` object containing:\cr"
  "- 'reference_time': The time point for which the growth rate is estimated.",
  "- 'observation': The observation in the reference time point.",
  "- 'season': The stratification of observables in corresponding seasons.",
  "- 'growth_rate': The estimated growth rate.",
  "- 'lower_growth_rate': The lower bound of the growth rate's confidence interval.",
  "- 'upper_growth_rate': The upper bound of the growth rate's confidence interval.",
  "- 'growth_warning': Logical. Is the growth rate significantly higher than zero?",
  "- 'sum_of_cases': The sum of cases within the time window.",
  "- 'sum_of_cases_warning': Logical. Does the Sum of Cases exceed the disease threshold?",
  "- 'seasonal_onset_alarm': Logical. Is there a seasonal onset alarm?",
  "- 'skipped_window': Logical. Was the window skipped due to missing?",
  "- 'converged': Logical. Was the IWLS judged to have converged?",
)
```

Here:
```
#' @return `r rd_season_onset_return`
```

In combined_seasonal_output.R
```
#' @return An object containing two lists: onset_output and burden_output:
#'
#' onset_output:
#' `r rd_season_onset_return`
#'
#' burden_output:
#' `r rd_burden_output_return`
```